core/src/main/java/org/apache/carbondata/core/datamap/DataMapUtil.java

```diff
@@ -113,7 +113,11 @@ private static void executeClearDataMapJob(DataMapJob dataMapJob,
     DistributableDataMapFormat dataMapFormat = new DistributableDataMapFormat(carbonTable,
         validAndInvalidSegmentsInfo.getValidSegments(), invalidSegment, true,
         dataMapToClear);
-    dataMapJob.execute(dataMapFormat);
+    try {
+      dataMapJob.execute(dataMapFormat);
```

Comment:
In this case fallback may not work, please check for it

Reply:
added check for DistributedJob